src/components/Posts/PostList.tsx

```diff
@@ -2,24 +2,49 @@ import { useEffect } from "react";
 import PostItem from "./PostItem";
 import axios from "axios";
 import { useAppDispatch, useAppSelector } from "../../state/hooks";
-import { Post, setPosts } from "../../state/slices/postSlice";
+import { Post, setPosts } from "../../state/slices/postsSlice";
 
-const Timeline = () => {
-  const posts = useAppSelector((state) => state.post);
+type PostListProps = {
+  replies: boolean;
+};
+
+const PostList = ({ replies }: PostListProps) => {
+  const posts = useAppSelector((state) => state.posts);
   const user = useAppSelector((state) => state.user);
+  const post = useAppSelector((state) => state.post);
   const dispatch = useAppDispatch();
 
   useEffect(() => {
     const fetchPosts = async () => {
-      const result = await axios.get("http://localhost:3001/api/posts", {
-        params: {
-          userId: user.userId,
-        },
-      });
-      dispatch(setPosts(result.data as Post[]));
+      try {
+        if (replies) {
```

Comment:
I think it's worth looking into how we can replace this `replies` prop with something better. In our profile page, we have 4 more post list components that have the same issue since all of our post lists are basically the same other than which endpoint they fetch data from

I feel like there must be a better way to determine which endpoint to use. Let's loop in @AnindaS13 too since he did the profile page post lists. Maybe we can let this slide in this PR and make a new ticket to figure this stuff out

Reply:
I agree with this, there needs to be a better way for us to use the PostList template. Seems like a lot of copy paste right now

Reply:
Yeah, I'm down for that. We'll make a new ticket to handle refactoring these post lists